tests/common/plugins/ptfadapter/ptfadapter.py

```diff
@@ -37,6 +52,15 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
         self.kill()
 
+    def _check_ptf_nn_agent_availability(self, socket_addr):
+        """Verify the nanomsg socket address exposed by ptf_nn_agent is available."""
+        sock = nnpy.Socket(nnpy.AF_SP, nnpy.PAIR)
+        sock.connect(socket_addr)
```

Comment:
I suggest setting the timeout for ```connect``` explicitly if possible. And do we need to ```get_statistic``` periodically? I think it's a oneshot behavior.

Reply:
Discussed offline, the reason to periodically retrieve established connection is that `nanomsg` needs some time to establish a tcp connection to the remote after `connect`.